oci/private/push.sh.tpl

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+set -o pipefail -o errexit -o nounset
+
+readonly CRANE="{{crane_path}}"
+readonly YQ="{{yq_path}}"
+readonly IMAGE_DIR="{{image_dir}}"
+readonly FIXED_ARGS=({{fixed_args}})
+
+# set $@ to be FIXED_ARGS+$@
+ALL_ARGS=(${FIXED_ARGS[@]} $@)
+set -- ${ALL_ARGS[@]}
+
+REPOSITORY="{{repository}}"
+TAGS=()
+ARGS=()
+
+while (( $# > 0 )); do
+  case $1 in
+    (-t|--tag)
```

Comment:
nit: ordering of the flag handling is reversed between `tag` and `repository` with `-t|--tag` and then `--tags=*` for tag and `--repository=*` and `-r|--repository` for repository